src/mscorlib/shared/System/Text/StringBuilder.cs

```diff
@@ -2125,8 +2086,8 @@ private StringBuilder FindChunkForByte(int byteIndex)
         /// </remarks>
         private void ExpandByABlock(int minBlockCharCount)
         {
-            Contract.Requires(Capacity == Length, $"{nameof(ExpandByABlock)} should only be called when there is no space left.");
-            Contract.Requires(minBlockCharCount > 0);
+            Debug.Assert(Capacity == Length, nameof(ExpandByABlock) + " should only be called when there is no space left.");
```

Comment:
Not a big deal, but I'm curious why you changed this away from using string interpolation?

Reply:
It causes a StringBuilder to be constructed, which ends up calling `ExpandByABlock`, causing a stack overflow. A nice reminder that this innocuous looking assert does a lot of work even when the condition is false.

Reply:
> It causes a StringBuilder to be constructed

That's something we should address separately.

Reply:
https://github.com/dotnet/roslyn/issues/22344